src/shell/commands/disk_rebalance.cpp

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) 2017, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "shell/commands.h"
+#include "shell/argh.h"
+
+#include <math.h>
+#include <fmt/ostream.h>
+#include <dsn/utility/errors.h>
+#include <dsn/utility/output_utils.h>
+#include <dsn/utility/string_conv.h>
+#include <dsn/dist/replication/duplication_common.h>
+
+bool validate_cmd(argh::parser &cmd,
+                  const std::set<std::string> &params,
+                  const std::set<std::string> &flags)
+{
+    if (cmd.size() > 1) {
+        fmt::print(stderr, "too many params!\n");
+        return false;
+    }
+
+    for (const auto &param : cmd.params()) {
+        if (std::find(params.begin(), params.end(), param.first) == params.end()) {
+            fmt::print(stderr, "unknown param {}={} \n", param.first, param.second);
+            return false;
+        }
+    }
+
+    for (const auto &flag : cmd.flags()) {
+        if (std::find(params.begin(), params.end(), flag) != params.end()) {
+            fmt::print(stderr, "missing value of {}\n", flag);
+            return false;
+        }
+
+        if (std::find(flags.begin(), flags.end(), flag) == flags.end()) {
+            fmt::print(stderr, "unknown flag {}\n", flag);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+std::ostream *get_out_stream(std::string &file_name)
+{
+    std::ostream *os_ptr = nullptr;
+    if (file_name.empty()) {
+        os_ptr = &std::cout;
+    } else {
+        os_ptr = new std::ofstream(file_name);
+        if (!*os_ptr) {
+            fmt::print(stderr, "ERROR: open output file {} failed\n", file_name);
+            delete os_ptr;
+        }
+    }
+    return os_ptr;
+}
+
+bool query_disk_info(
+    shell_context *sc,
+    argh::parser &cmd,
+    std::string &node_address,
```

Comment:
All parameters passed by lvalue reference must be labeled const.
--https://google.github.io/styleguide/cppguide.html#Reference_Arguments